### DISCUSSION TITLE
When will there be aot support for x86 (32 bit)?

### DISCUSSION THREAD

**Discussion by CeSun** [dotnet/runtime#77646] [Q&A] [answered]
Will the .net7 release version be supported?

**Comment by huoyaoyuan** [comment]
See https://github.com/dotnet/runtimelab/issues/1471

The ABI and calling convention on x86 is more complicated than other platforms.

**Reply by CeSun** [reply]
Thank you😊.

**Reply by Joe4evr** [reply]
For the record: https://devblogs.microsoft.com/oldnewthing/20040914-00/?p=37873
https://devblogs.microsoft.com/oldnewthing/20220418-00/?p=106489

**Comment by CeSun** [comment, marked as answer]
@kant2002 Does NativeAOT support x86 and merge it into the main branch? Can the .net7 rc2 version be used?
 https://github.com/dotnet/runtime/pull/68130


I wanted to say "excuse me / Please", but my translation software didn't translate it, the tone was a bit offensive, sorry.

**Reply by kant2002** [reply]
Hi @CeSun NativeAOT does not supported x86, and it's quite a lot of work to make that happens. I do not have time for implementing that now. Can you tell, why you cannot use regular CoreCLR x86 for your project?

> I wanted to say "excuse me / Please", but my translation software didn't translate it, the tone was a bit offensive, sorry.

Don't worry, everything fine, English is not my first language either, so no offence.

**Reply by CeSun** [reply]
> Hi @CeSun NativeAOT does not supported x86, and it's quite a lot of work to make that happens. I do not have time for implementing that now. Can you tell, why you cannot use regular CoreCLR x86 for your project?
> 
> > I wanted to say "excuse me / Please", but my translation software didn't translate it, the tone was a bit offensive, sorry.
> 
> Don't worry, everything fine, English is not my first language either, so no offence.

Integrating clr is more cumbersome, and distributing my software requires the user to install the runtime, which is somewhat unacceptable to me.

If NativeAOT really can't solve it at all, I can only access clr or mono

**Reply by kant2002** [reply]
Are you considering self containing deployment and trimming? It may create large size of the app, but that way you just ask user to install your app and that’s it. Trimming a bit problematic, but same issues you would see with NativeAOT so maybe that s acceptable for you.

**Reply by CeSun** [reply]
> Are you considering self containing deployment and trimming? It may create large size of the app, but that way you just ask user to install your app and that’s it. Trimming a bit problematic, but same issues you would see with NativeAOT so maybe that s acceptable for you.

Ok thank you i'll try it.
